src/Microsoft.AspNetCore.Mvc.Razor/Compilation/IViewCompilerProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public interface IViewCompilerProvider
```

Comment:
Yeah I think this is fine. The other option is to just use a factory method with DI.

The problem with either approach is that we need to be careful to lazy-initialize this **everywhere** or else it doesn't add value. I think what you've done here is fine.